anyway/widgets/widget_utils.py

```diff
@@ -128,3 +128,29 @@ def format_2_level_items(
             series_data.append({LKEY: l2, VAL: num})
         res.append({LKEY: l1, SERIES: series_data})
     return res
+
+
+def order_by_list_stack_bar_widget(
+    structured_data_list: List[Dict[str, Union[str, List]]], severity_order: List[str]
+) -> List[Dict[str, Union[str, List]]]:
+    severity_dict = {}
+    for struct in structured_data_list:
+        series_data = struct["series"]
+        for injury in series_data:
+            for key, value in injury.items():
+                if key != "label_key":
+                    continue
+
+                severity_dict[value] = injury
+
+        series_data_new = []
+        for severity in severity_order:
+            if severity in severity_dict:
+                series_data_new.append(severity_dict.pop(severity))
+
+        for key, value in severity_dict.items():

```

Comment:
you popped all the items in the previous iteration over severity_dict, so this enumeration will be empty no?

Reply:
In the current use case it will pop all the items in the dict, but i am planning on using this function in other cases (We need to sort 4 other widgets) so i don't want to rely on that (that all the keys in the dict are in `severity_order)`